app/src/common/chromium/com/igalia/wolvic/browser/api/impl/PromptDelegateImpl.java

```diff
@@ -823,4 +833,239 @@ public WSession.PromptDelegate.PromptResponse dismiss() {
+                    options[0] = new WAutocomplete.LoginSaveOption(fromPasswordForm(form));
+                    mLoginSavePrompt = new LoginSavePrompt(mPasswordManagerListener, options);
+                    ThreadUtils.postOnUiThread(() -> {
+                        try {
+                            if (mDelegate != null) {
+                                mDelegate.onLoginSave(mSession, mLoginSavePrompt);
+                            } else {
+                                resetLoginSavePrompt();
+                            }
+                        } catch (WindowManager.BadTokenException e) {
+                            resetLoginSavePrompt();
+                        }
+                    });
+                }
+                return null;
+            });
+
+            return true;
+        }
+
+        @Override
+        public boolean onLoginSelect(PasswordForm[] forms) {
+            dismiss();
+            assert mPasswordManagerListener != null;
+
+            WAutocomplete.LoginSelectOption[] options =
+                    new WAutocomplete.LoginSelectOption[forms.length];
+            for (int i = 0; i < forms.length; i++) {
+                options[i] = new WAutocomplete.LoginSelectOption(fromPasswordForm(forms[i]));
+            }
+
+            mLoginSelectPrompt = new LoginSelectPrompt<PasswordManager.Listener>(
+                    mPasswordManagerListener, options);
+            try {
+                if (mDelegate != null) {
+                    mDelegate.onLoginSelect(mSession, mLoginSelectPrompt);
+                } else {
+                    resetLoginSelectPrompt();
+                    return false;
+                }
+            } catch (WindowManager.BadTokenException e) {
+                resetLoginSelectPrompt();
+                return false;
+            }
+            return true;
+        }
+
+        @Override
+        public boolean onLoginSelect(String[] username) {
+            dismiss();
+            assert mAutofillManagerListener != null;
+
+            WAutocomplete.LoginSelectOption[] options =
+                    new WAutocomplete.LoginSelectOption[username.length];
+            for (int i = 0; i < username.length; i++) {
+                options[i] = new WAutocomplete.LoginSelectOption(fromUsername(i, username[i]));
+            }
+
+            mLoginSelectPrompt = new LoginSelectPrompt<AutofillManager.Listener>(
+                    mAutofillManagerListener, options);
+            try {
+                if (mDelegate != null) {
+                    mDelegate.onLoginSelect(mSession, mLoginSelectPrompt);
+                } else {
+                    resetLoginSelectPrompt();
+                    return false;
+                }
+            } catch (WindowManager.BadTokenException e) {
+                resetLoginSelectPrompt();
+                return false;
+            }
+            return true;
+        }
```

Comment:
This method is nearly identical to the above one, can't we factorize them in a private method? Maybe not the whole method but at least starting from the `mLoginSelectPrompt`  assignment it should be quite easy to refactor.

Reply:
Done